profiler/models/profiler_profile.py

```diff
@@ -0,0 +1,470 @@
+
+    @api.model
+    def _find_loggers_path(self):
+        self.env.cr.execute("SHOW log_directory")
+        log_directory = self.env.cr.fetchone()[0]
+        self.env.cr.execute("SHOW log_filename")
+        log_filename = self.env.cr.fetchone()[0]
+        log_path = os.path.join(log_directory, log_filename)
+        if not os.path.isabs(log_path):
+            # It is relative path then join data_directory
+            self.env.cr.execute("SHOW data_directory")
+            data_dir = self.env.cr.fetchone()[0]
+            log_path = os.path.join(data_dir, log_path)
+        return log_path
+
+    name = fields.Char(required=True)
+    enable_python = fields.Boolean(default=True)
+    enable_postgresql = fields.Boolean(
+        default=False,
+        help="It requires postgresql server logs seudo-enabled")
+    use_py_index = fields.Boolean(
+        name="Get cProfiling report", default=False,
+        help="Index human-readable cProfile attachment."
+        "\nTo access this report, you must open the cprofile attachment view "
+        "using debug mode.\nWarning: Uses more resources.")
+    date_started = fields.Char(readonly=True)
+    date_finished = fields.Char(readonly=True)
+    state = fields.Selection([
+        ('enabled', 'Enabled'),
+        ('disabled', 'Disabled'),
+    ], default='disabled', readonly=True, required=True)
+    description = fields.Text(readonly=True)
+    attachment_count = fields.Integer(compute="_compute_attachment_count")
+    pg_log_path = fields.Char(help="Getting the path to the logger",
+                              default=_find_loggers_path)
+    pg_remote = fields.Char()
+    pg_stats_slowest_html = fields.Html(
+        "PostgreSQL Stats - Slowest", readonly=True)
+    pg_stats_time_consuming_html = fields.Html(
+        "PostgreSQL Stats - Time Consuming", readonly=True)
+    pg_stats_most_frequent_html = fields.Html(
+        "PostgreSQL Stats - Most Frequent", readonly=True)
+    py_stats_lines = fields.One2many(
+        "profiler.profile.python.line", "profile_id", "PY Stats Lines")
+
+    @api.multi
+    def _compute_attachment_count(self):
+        for record in self:
+            self.attachment_count = self.env['ir.attachment'].search_count([
+                ('res_model', '=', self._name), ('res_id', '=', record.id)])
+
+    @api.onchange('enable_postgresql')
+    def onchange_enable_postgresql(self):
+        if not self.enable_postgresql:
+            return
+        self.env.cr.execute("SHOW config_file")
+        pg_config_file = self.env.cr.fetchone()[0]
+        db_host = tools.config.get('db_host')
+        if db_host == 'localhost' or db_host == '127.0.0.1':
+            db_host = False
+        if db_host:
+            pg_config_file = 'postgres@%s:%s' % (db_host, pg_config_file)
+            self.pg_remote = db_host
+
+        self.description = (
+            "You need seudo-enable logs from your "
+            "postgresql-server configuration file.\n"
```

Comment:
Any chance for this very addon to enable those options? Using https://www.postgresql.org/docs/current/static/sql-set.html or similar?